### PR TITLE
`Exam mode`: Update submitted and started exam details in real time

### PR DESCRIPTION
**Pull request by Dominik-Weinzierl**
#### General

- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

@krusche asked me to implement this functionality as my final task. This will allow instructors to stay on the exam configuration page without reloading.

### Description

- We send the following messages once a student submits (`/topic/exam/${exam.id}/submitted`) or starts (`/topic/exam/${exam.id}/started`) his exam

### Steps for Testing

Prerequisites:
- 1 Instructor
- Multiple Students
- Exam

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an Exam
4. Register multiple students
5. Participate with those students and keep the `Exam Management` page open (see screenshots)
6. You should notice that the statistics are updated (e.g. started and submitted student)
7. This also happens for the exam checklist (see screenshots)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

Tests unchanged and won`t be added in this PR.

### Screenshots

<details>
<summary>Testing Screenshots</summary>

image
image
</details>

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java
Comment (ge65cer) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:286]: Can this even be `null`?
```suggestion
        if (!studentExam.isStarted()) {
```

File: src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html
Comment (ge65cer) [inline src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html:293]: `numberOfStarted` can not be undefined, because you already test for `> 0` above.
Please review the other occurrences as well.
```suggestion
                                [percentage]="(100 * numberOfStarted) / examChecklist.numberOfGeneratedStudentExams!"
```

**Review by ge65cer — CHANGES_REQUESTED** [review]
Code reviewed. We should test this in a testing session…

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java
Comment (Dominik-Weinzierl) [reply src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:286]: Yes, it can.

File: src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html
Comment (Dominik-Weinzierl) [reply src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html]: Sure, done 👍

File: src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html
Comment (Dominik-Weinzierl) [reply src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html]: Thanks!

**Review by ge65cer — DISMISSED** [review]
Thanks for incorporating the feedback. I tested the PR on TS1 and everything worked without issues.

I was a little confused that the "Submitted" progress bar is relative to the number of started exams, whereas on the Exam Management page the "Submitted" percentage is based on total number of registered students. You could add an explanation popup, but let's see what the other reviewers say about that.

**Comment by Dominik-Weinzierl** [comment]
> You could add an explanation popup, but let's see what the other reviewers say about that.

Thanks for your testing! Yeah, we can create an issue for that if it is really required 👍

**Comment by Dominik-Weinzierl** [comment]
I mark this PR as ready to merge and ready for review - so additional reviews are highly appreciated, but not mandatory :+1:

**Review by stefanwaldhauser — APPROVED** [review]
Works great and the code looks good to! GJ! 👍

**Review by ge65cer — APPROVED** [review]
Re-approve after merge.

**Pull request merged**
